### PR TITLE
New Tutorial: Kotlin Spring Boot on Google App Engine

### PR DESCRIPTION
**Pull request by hhariri**
- [x] Tech writing / styleguide review - @steveperry-53  
- [x] Formatting review for publishing - @jmdobry

### CODE REVIEW

File: tutorials/kotlin-springboot-app-engine.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+and you need to have Docker and the Google Cloud SDK installed.
+
+Create a project that will host your Spring Boot application. You can also reuse
+an existing project.
+
+1.  Use the [Google Cloud Platform Console](https://console.cloud.google.com/)
+    to create a new Cloud Platform project. Remember the project ID; you will
+    need it later. Later commands in this tutorial will use `${PROJECT_ID}` as
+    a substitution, so you might consider setting the `PROJECT_ID` environment
+    variable in your shell.
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: It looks like the later commands use [PROJECT_ID], not ${PROJECT_ID}.

File: tutorials/kotlin-springboot-app-engine.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-starter-web</artifactId>
+    <version>RELEASE</version>
+</dependency>
+```
+
+5. Run the application from the command line using Maven:
+
+    mvn spring-boot:run
+
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: Use ``` around this command.

File: tutorials/kotlin-springboot-app-engine.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+6. Open the browser and make sure your get a valid JSON response when accessing http://localhost:8080/message. The result should be:
+
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: your get -> you get

File: tutorials/kotlin-springboot-app-engine.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+@RestController
+class MessageController {
+    @RequestMapping("/message")
+    fun message(): Message {
+        return Message("Hello from Google Cloud", "High")
+    }
+}
+```
+
+The package should match that of your group and artifact name.
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: The file contents and following sentence are not getting indented as part of step 3. Could you try ```shell to see if that works better. Or you could indent 8 spaces.

File: tutorials/kotlin-springboot-app-engine.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+
+6. Open the browser and make sure your get a valid JSON response when accessing http://localhost:8080/message. The result should be:
+
+```json
+{
+"text": "Hello from Google Cloud",
+"priority": "High"
+}
+```
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: The json block is not getting indented as part of step 6. Try ```shell or indenting 8 spaces.

File: tutorials/kotlin-springboot-app-engine.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+2.  Add the following contents to the `pom.xml` file to configure the Maven plugin:
+
+```xml
+<plugin>
+    <groupId>com.google.cloud.tools</groupId>
+    <artifactId>appengine-maven-plugin</artifactId>
+    <version>1.2.1</version>
+</plugin>
+```
+
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: XML is not getting indented as part of step 2.

File: tutorials/kotlin-springboot-app-engine.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    **Note**: If the command fails with `Google Cloud SDK could not be found`, make sure the environment
+    variable `GOOGLE_CLOUD_SDK_HOME` is set to the root directory of where you installed the Google Cloud SDK. 
+    
+4.  Once the deploy command has completed, you can run
+
+        gcloud app browse
+
+    to see your app running in production on App Engine in the browser. Not however that this application does not
+    respond to the root endpoint. Once the browser is open with the correct URL, you need to suffix `/message` to it.
+
```
Comment (steveperry-53) [inline tutorials/kotlin-springboot-app-engine.md]: suffix -> append

File: tutorials/kotlin-springboot-app-engine.md (hunk 4 above)
Comment (hhariri) [reply tutorials/kotlin-springboot-app-engine.md]: Do you mean the Kotlin code? It's using three ticks with Kotlin...

**Comment by hhariri** [comment]
Not sure why the indentation isn't working. I've put tabs but left the ticks and the proper language type....

**Comment by steveperry-53** [comment]
@hhariri Yeah, I find indentation baffling. At any rate, the indentation is much better now, and I think the topic is ready to publish.

**Pull request merged**